### PR TITLE
Merge ObjectAdapterI into ObjectAdapter in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR merges ObjectAdapterI into ObjectAdapter + various other clean-ups in C#.

### CODE REVIEW

File: csharp/src/Ice/ObjectAdapter.cs [hunk 1]
```diff
@@ -2,29 +2,140 @@
...
+using Ice.Instrumentation;
+using Ice.Internal;
+using System.Diagnostics;
+using System.Net.Security;
+using System.Text;
+
 namespace Ice;
 
-public interface ObjectAdapter
+public sealed class ObjectAdapter
```
Comment (bernardnormier) [inline csharp/src/Ice/ObjectAdapter.cs:13]: Essentially moved all the code from ObjectAdapterI in here + cleanup + nullable enable.

File: csharp/src/Ice/Internal/EndpointI.cs [hunk 2]
```diff
@@ -13,11 +13,6 @@ public interface EndpointI_connectors
 public abstract class EndpointI : Ice.Endpoint, IComparable<EndpointI>
 {
     public override string ToString()
-    {
-        return ice_toString_();
-    }
-
-    public virtual string ice_toString_()
```
Comment (bernardnormier) [inline csharp/src/Ice/Internal/EndpointI.cs:20]: Also removed all the `ice_toString_` methods. You should obviously use ToString()!

**Review by externl — APPROVED** [review]

File: csharp/src/Ice/Internal/ObjectAdapterFactory.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@ public sealed class ObjectAdapterFactory
 {
     public void shutdown()
     {
-        List<Ice.ObjectAdapterI> adapters;
+        List<Ice.ObjectAdapter> adapters;
```
Comment (pepone) [inline csharp/src/Ice/Internal/ObjectAdapterFactory.cs:10]: We should clean-up Ice. qualification in this file.

**Review by pepone — APPROVED** [review]

File: csharp/src/Ice/Internal/ObjectAdapterFactory.cs (hunk 3 above)
Comment (bernardnormier) [reply csharp/src/Ice/Internal/ObjectAdapterFactory.cs:10]: And many other files. That's for another PR.

**Pull request merged**